core/modules/parser/QSMySqlListener.cc

```diff
@@ -1934,10 +1935,15 @@ class LogicalExpressionAdapter :
 
         case LogicalOperatorCBH::AND:
             // We capture the AndTerm into a base class so we can pass by reference into the setter.
-            shared_ptr<query::LogicalTerm> logicalTerm = make_shared<query::AndTerm>();
-            _setLogicalOperator(logicalTerm);
+            _setLogicalOperator(make_shared<query::AndTerm>());
+            break;
+
+        case LogicalOperatorCBH::OR: {
```

Comment:
Small style inconsistency - previous `case` does not use block `{}` but this one does, and I don't see much of a difference in the content which needs a block here.